salt/payload.py

```diff
@@ -132,9 +133,11 @@ def loads(self, msg, encoding=None):
                 # Due to this, if we don't need it, don't pass it at all so
                 # that under Python 2 we can still work with older versions
                 # of msgpack.
-                return msgpack.loads(msg, use_list=True, encoding=encoding)
+                ret = msgpack.loads(msg, use_list=True, encoding=encoding)
             else:
-                return msgpack.loads(msg, use_list=True)
+                ret = msgpack.loads(msg, use_list=True)
+            if six.PY3 and not raw:
+                ret = salt.transport.frame.decode_embedded_strs(ret)
```

Comment:
Also, as for the tcp transport, it calls msgpack directly (which it should not do) so it would still need to decode, I will look at cleaning that up in another PR